base/range.jl

```diff
@@ -473,11 +473,23 @@ end
 
 ## indexing
 
+_in_unit_range(v::UnitRange, val, i::Integer) = i > 0 && val <= v.stop && val >= v.start
+
 function getindex(v::UnitRange{T}, i::Integer) where T
     @_inline_meta
-    ret = convert(T, first(v) + i - 1)
-    @boundscheck ((i > 0) & (ret <= v.stop) & (ret >= v.start)) || throw_boundserror(v, i)
-    ret
+    val = convert(T, v.start + i - 1)
+    @boundscheck _in_unit_range(v, val, i) || throw_boundserror(v, i)
+    val
+end
+
+const OverflowSafe = Union{Bool,Int8,Int16,Int32,Int64,Int128,
+                           UInt8,UInt16,UInt32,UInt64,UInt128}
+
+function getindex(v::UnitRange{T}, i::Integer) where {T<:OverflowSafe}
```

Comment:
You could just use `Union{Bool, Base.BitInteger}` for this.

Reply:
Yeah, I was trying that, but it turns out `BitInteger` is defined a couple includes away in `int.jl`, and when I tried moving the constants to a separate file `int_types.jl`, I broke stuff without very useful feedback from `make`. It's worth defining these types a bit earlier, because other places redefine them as well!